### ISSUE TITLE
TFRecords preprocessing

### ISSUE THREAD

**Issue by bluestyle97** [Andy97/DeepMLS#3]
Hi, I want to generate tfrecords training data from another mesh dataset (which has already been watertight), but there seems to be no instruction on this step. I have read [this istruction](https://github.com/Andy97/DeepMLS/tree/master/vdb_tsdf), but the code seems to compute sdf values only and have no help with my question. I found there are fields in the tfrecords data that I don't know what they means and where they come from, such as "sdf_samples_depth6", "sdf_samples_depth7", "depth6_coord" and "depth7_coord". Can you give a more detailed instruction on how to generate the right-format tfrecords data? Thanks a lot!

**Comment by Andy97** [comment]
> Hi, I want to generate tfrecords training data from another mesh dataset (which has already been watertight), but there seems to be no instruction on this step. I have read [this istruction](https://github.com/Andy97/DeepMLS/tree/master/vdb_tsdf), but the code seems to compute sdf values only and have no help with my question. I found there are fields in the tfrecords data that I don't know what they means and where they come from, such as "sdf_samples_depth6", "sdf_samples_depth7", "depth6_coord" and "depth7_coord". Can you give a more detailed instruction on how to generate the right-format tfrecords data? Thanks a lot!

Hi, as mentioned in the paper, we have 2 set of sdf samples(sampled from 64x64x64 grid points and 128x128x128 grid points, please refer to Appendix A of paper), which were used for sdf training in a coarse-to-fine manner progressively(e.g. using 64^3 samples for first 30 epochs and 128^3 samples for remaining epochs).  The "depth6_coord" and "depth7_coord" is the 3d coordinates of these sample points, while "sdf_samples_depth6", "sdf_samples_depth7" contains the corresponding ground truth sdf values and gradients. For details, please refer to [code for data loading](https://github.com/Andy97/DeepMLS/blob/master/points3d-tf/pointsdataset/points_dataset_octreed7.py#L36)

**Comment by Andy97** [comment]
Script for generating tfrecords is released, please see [here](https://github.com/Andy97/DeepMLS/tree/master/scripts).
